test/test_datapipe.py

```diff
@@ -2015,19 +2015,23 @@ class _CustomIterDataPipe(IterDataPipe):
         def noop(self, x):
             return x + 1
 
-        def __init__(self):
-            self._dp = dp.iter.IterableWrapper([1, 2, 4]).map(self.noop)
+        def __init__(self, fn):
+            self.fn = fn
+            self._dp = dp.iter.IterableWrapper([1, 2, 4]).map(self.noop).map(self.fn)
 
         def __iter__(self):
             yield from self._dp
 
     def test_circular_serialization_with_pickle(self):
-        assert list(self._CustomIterDataPipe()) == list(pickle.loads(pickle.dumps(self._CustomIterDataPipe())))
+        assert list(self._CustomIterDataPipe(_fake_fn)) ==\
+               list(pickle.loads(pickle.dumps(self._CustomIterDataPipe(_fake_fn))))
 
-    # TODO: Ensure this works with `dill` installed
-    # @skipIfNoDill
-    # def test_circular_serialization_with_dill(self):
-    #     assert list(self._CustomIterDataPipe()) == list(dill.loads(dill.dumps(self._CustomIterDataPipe())))
+    @skipIfNoDill
+    def test_circular_serialization_with_dill(self):
```

Comment:
I am open to suggestion here to make these test cases more extensive.

Reply:
Except lambda, can you add a local function here for testing?
```
def test_circular_serialization_with_dill(self):
    def _local_fn(x):
         return x + 1
    list(self._CustomIterDataPipe(_local_fn))